scalyr_agent/monitor_utils/tests/k8s_test.py

```diff
@@ -513,6 +514,68 @@ def test_no_prefetch(self):
         self.assertEqual(10, value)
 
 
+class TestK8sNamespaceFilter(ScalyrTestCase):
+    """Tests the DockerMetricFetch abstraction.
+    """
+
+    def test_basic_blacklist(self):
+        test_filter = K8sNamespaceFilter(global_include=["*"], global_ignore=["kube"])
+        self.assertTrue(test_filter.passes("scalyr"))
+        self.assertFalse(test_filter.passes("kube"))
+        self.assertTrue("scalyr" in test_filter)
+        self.assertTrue("kube" not in test_filter)
+
+        test_filter = K8sNamespaceFilter(
+            global_include=["*"], global_ignore=["kube", "test"]
+        )
+        self.assertTrue(test_filter.passes("scalyr"))
+        self.assertFalse(test_filter.passes("kube"))
+        self.assertFalse(test_filter.passes("test"))
+
+    def test_basic_whitelist(self):
+        test_filter = K8sNamespaceFilter(global_include=["scalyr"], global_ignore=[])
+        self.assertTrue(test_filter.passes("scalyr"))
+        self.assertFalse(test_filter.passes("kube"))
+
+        test_filter = K8sNamespaceFilter(
+            global_include=["scalyr", "your-app"], global_ignore=[]
+        )
+        self.assertTrue(test_filter.passes("scalyr"))
+        self.assertTrue(test_filter.passes("your-app"))
+        self.assertFalse(test_filter.passes("kube"))
+
+    def test_basic_whitelist_and_blacklist(self):
+        test_filter = K8sNamespaceFilter(
+            global_include=["scalyr", "your-app", "kube"],
+            global_ignore=["kube", "testing"],
+        )
+        self.assertTrue(test_filter.passes("scalyr"))
+        self.assertTrue(test_filter.passes("your-app"))
+        self.assertFalse(test_filter.passes("kube"))
```

Comment:
For completeness, do we also want to test `self.assertFalse(test_filter.passes("testing"))`

Reply:
Done.